lib/elastic_apm/opentracing.rb

```diff
@@ -0,0 +1,242 @@
+# frozen_string_literal: true
+
+require 'opentracing'
+
+module ElasticAPM
+  module OpenTracing
+    # @api private
+    class Span
+      def initialize(elastic_span)
+        @elastic_span = elastic_span
+      end
+
+      attr_reader :elastic_span
+
+      def context
+        elastic_span.trace_context
+      end
+
+      def finish(end_time: Time.now)
+        elastic_span.done end_time: Util.micros(end_time)
+
+        case elastic_span
+        when ElasticAPM::Transaction
+          ElasticAPM.end_transaction
+        when ElasticAPM::Span
+          ElasticAPM.end_span
```

Comment:
Yes, this is also the comment I made on Slack. It's the weirdest part.

I think we could maybe run through `current_spans` and find the correct one. But that's just throwing more weirdness and n+1 in there.

If we just say that `ignore_active_scope` is unsupported I think this would _always_ work?

Reply:
In theory the users could still end a dedicated span at any time, where the ruby agent would then end the current span. Did you check with the other agent devs if someone is in a similar situation?

Reply:
I've updated my approach to mirror the logic from instrumenter, sort of. Curious what you think.